lib/adiwg/mdtranslator/readers/iso19115_2/modules/module_maintenance.rb

```diff
@@ -0,0 +1,36 @@
+require 'nokogiri'
+require 'adiwg/mdtranslator/internal/internal_metadata_obj'
+
+module ADIWG
+   module Mdtranslator
+      module Readers
+         module Iso191152
+            module Maintenance
+               @@maintenanceInfoXPath = 'gmd:MD_MaintenanceInformation'
+               @@frequencyXPath = 'gmd:maintenanceAndUpdateFrequency/gmd:MD_MaintenanceFrequencyCode'
+               def self.unpack(xMaintenance, _hResponseObj)
+                  intMetadataClass = InternalMetadata.new
+                  hMaintenance = intMetadataClass.newMaintenance
+
+                  xMaintenanceInfo = xMaintenance.xpath(@@maintenanceInfoXPath)[0]
+                  return nil if xMaintenanceInfo.nil?
+
+                  # only thing dcatus needs
+                  # :frequency (optional)
+                  # <xs:element name="maintenanceAndUpdateFrequency" type="gmd:MD_MaintenanceFrequencyCode_PropertyType"/>
```

Comment:
this is a required property. please change the comment to reflect that. also, because it's required a warning needs to be issued when it's missing. for example, 
```ruby
if xFrequency.nil? 
  msg = "WARNING: ISO19115-2 reader: element '#{@@frequencyXPath}' " \
  "is missing in #{@@maintenanceInfoXPath}"
  hResponseObj[:readerExecutionMessages] << msg
  hResponseObj[:readerExecutionPass] = false
end
```
you're also missing a test module for `Maintenance` and a `hResponseObj[:readerExecutionMessages]` test verifying the appropriate warning was thrown when the property was missing. you'll want to create a fixture with `maintenanceAndUpdateFrequency` missing.

Reply:
made it a required now.